Azure-Samples/azure-functions-code-testing-sample
# Feature/improve integration testing infrastructure

## Purpose
* Added TestContainers and WireMock for realistic integration testing in the real world.
* Replaced Entity Framework with Dapper and CosmosDb with Postgresql. Added Bogus into the mix for realistic fake data generation.
* Update README to reflect the new changes and improve the style and structure of it.

## Does this introduce a breaking change?
```
[] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[] Bugfix
[x] Feature
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[] Other... Please describe:
```

## How to Test
* Get the code by cloning the repo
* Run all the tests of the solution in Visual Studio
* If you want to be able to run an instance of the API locally and test its full functionality make sure to see the README file and follow the steps described in there.

## What to Check
Verify that the following are valid
* Unit and integration tests are passing
